substratest/assets.py

```diff
@@ -131,69 +129,26 @@ class _ComputePlanFutureMixin(_BaseFutureMixin):
     _future_cls = ComputePlanFuture
 
 
-def _convert(name):
-    """Convert camel case to snake case."""
-    s = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', name)
-    return re.sub('([a-z0-9])([A-Z])', r'\1_\2', s).lower()
+_MANUAL_CONVERTION_FIELDS = {
+    'authorizedIDs': 'authorized_ids',
+    'dataManagerKey': 'dataset_key',
+    'pkhash': 'key',
+}
 
 
-class _DataclassLoader(abc.ABC):
-    """Base model structure defining assets.
+def convert_asset_field_names(name):
+    """Converts asset camel case fields to snake case fields.
 
-    Provides a load method to create the object from a dictionary.
-    Converts automatically camel case fields to snake case fields and provides a mapper
-    to define custom fields mapping.
+    Special cases are handled through a global dict variable.
     """
+    # XXX using a mapper for converting specific is not very flexible as it will be
+    #     applied to all fields from all assets.
+    mapper = _MANUAL_CONVERTION_FIELDS
```

Comment:
Because _MANUAL_CONVERTION_FIELDS is only used here, maybe we could move its content directly in `mapper`?

Reply:
It's a good practise to define **constant** variables as global variables.

Reply:
Okay 👍